include/ddc/kernels/splines/matrix.hpp

```diff
@@ -47,44 +48,22 @@ class Matrix
 
     virtual ddc::DSpan1D solve_inplace(ddc::DSpan1D const b) const
     {
-        assert(int(b.extent(0)) == m_n);
-        int const info = solve_inplace_method(b.data_handle(), 'N', 1);
-
-        if (info < 0) {
-            std::cerr << -info << "-th argument had an illegal value" << std::endl;
-            // TODO: Add LOG_FATAL_ERROR
-        }
+        ddc::DSpan2D_left b_2d(b.data_handle(), b.extent(0), 1);
+        solve_inplace(b_2d);
         return b;
     }
 
     virtual ddc::DSpan1D solve_transpose_inplace(ddc::DSpan1D const b) const
     {
-        assert(int(b.extent(0)) == m_n);
-        int const info = solve_inplace_method(b.data_handle(), 'T', 1);
-
-        if (info < 0) {
-            std::cerr << -info << "-th argument had an illegal value" << std::endl;
-            // TODO: Add LOG_FATAL_ERROR
-        }
+        ddc::DSpan2D_left b_2d(b.data_handle(), b.extent(0), 1);
+        solve_transpose_inplace(b_2d);
         return b;
     }
 
-    virtual ddc::DSpan2D solve_multiple_inplace(ddc::DSpan2D const bx) const
+    virtual ddc::DSpan2D_stride solve_inplace(ddc::DSpan2D_stride const bx) const
```

Comment:
What is the difference between `DSpan2D_left` and `DSpan2D_stride`?

Reply:
The same between Kokkos::LayoutLeft and Kokkos::LayoutStride. LayoutLeft is necessarily contiguous in left dimension and forbid `extent!=stride` in the right dimension.